### PR TITLE
Upgrade contract-proxy-kit and TS

### LINKED ISSUE #1033: gas limit estimations are way off
I am currently trying to redeem winnings for market:
https://omen.eth.link/#/0xe17a114ebe42ee2065fdc642398fc0e993142b4a

it looks like the gas limit estimations are way to high:
Screenshot 2020-08-03 at 15 11 47

### PR DESCRIPTION
**Pull request by cag**
Had to upgrade TypeScript in order to upgrade the CPK. Anyway, I made a dumb error, so the stuff had nothing to do with throttling: https://github.com/gnosis/contract-proxy-kit/pull/109

Resolves #1033

### CODE REVIEW

File: app/src/components/button/button_styling_types.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ const getButtonTypeStyles = (buttonType: ButtonType = ButtonType.primaryLine): a
   return PrimaryCSS
 }
 
-export const ButtonCSS = css<ButtonProps>`
```
Comment (cag) [inline app/src/components/button/button_styling_types.ts:132]: New TS didn't like it when this was being used with ButtonLinkProps.

File: app/src/components/market/common/customizable_tokens_select/modal_collateral.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const ModalCollateral = (props: Props) => {
   const [collateralAddress, setCollateralAddress] = useState<string>('')
   const { collateral, errorMessage, isSpinnerOn } = useCollateral(collateralAddress, context)
 
-  const validCollateralAddress: Maybe<Token> = collateral ? { address: collateralAddress, ...collateral } : null
+  const validCollateralAddress: Maybe<Token> = collateral ? { ...collateral } : null
```
Comment (cag) [inline app/src/components/market/common/customizable_tokens_select/modal_collateral.tsx:51]: TS says that that property will always get overwritten.

File: app/src/components/market/common/history_chart/chart.tsx [hunk 3]
```diff
@@ -110,7 +110,7 @@ const Legend = styled.li`
   }
 `
 
-const AnEvenSmallerLittleBall = styled(OutcomeItemLittleBallOfJoyAndDifferentColors)`
+const AnEvenSmallerLittleBall = styled(OutcomeItemLittleBallOfJoyAndDifferentColors as any)`
```
Comment (cag) [inline app/src/components/market/common/history_chart/chart.tsx:113]: Errors about this being a reference loop in the compilation occur seemingly randomly with the new TS (passing in `styled(styled(stuff))` causes some compilation to fail sometimes...)

File: app/src/hooks/useCpk.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import CPK from 'contract-proxy-kit'
-import EthersAdapter from 'contract-proxy-kit/lib/cjs/eth-lib-adapters/EthersAdapter'
+import EthersAdapter from 'contract-proxy-kit/lib/esm/ethLibAdapters/EthersAdapter'
```
Comment (cag) [inline app/src/hooks/useCpk.tsx:2]: Webpack actually uses the ESM versions when importing from 'contract-proxy-kit', so this change will make everything more consistent.

File: app/src/components/market/common/history_chart/chart.tsx (hunk 3 above)
Comment (cag) [reply app/src/components/market/common/history_chart/chart.tsx:113]: Okay probably should `s/styled\((\w+) as any\)/styled($1)` after upgrading styled-components is the proper fix here... https://github.com/microsoft/TypeScript/issues/37597

File: app/src/components/button/button_styling_types.ts (hunk 1 above)
Comment (cag) [reply app/src/components/button/button_styling_types.ts:132]: Brownie points if you can figure out the right type arg there.

File: app/src/components/market/common/history_chart/chart.tsx (hunk 3 above)
Comment (KadenZipfel) [reply app/src/components/market/common/history_chart/chart.tsx:113]: Should we add the `styled-components.d.ts` file to the project directly, as per: https://github.com/microsoft/TypeScript/issues/37597#issuecomment-628149946?

**Review by KadenZipfel — APPROVED** [review]
LGTM. Just left a question about types with styled-components

**Review by pimato — APPROVED** [review]

File: app/src/components/market/common/history_chart/chart.tsx (hunk 3 above)
Comment (cag) [reply app/src/components/market/common/history_chart/chart.tsx:113]: I would check if the new type definitions have been merged in before before correcting the issue.

**Pull request merged**